code/modules/mob/living/living_say.dm

```diff
@@ -254,6 +254,12 @@ GLOBAL_LIST_INIT(message_modes_stat_limits, list(
 	if(pressure < ONE_ATMOSPHERE*0.4) //Thin air, let's italicise the message
 		spans |= SPAN_ITALICS
 
+	if(!(SPAN_COMMAND in spans))
+		var/obj/structure/closet/crate/soapbox = locate(/obj/structure/closet/crate) in T
+		if(soapbox && !soapbox.opened && !ismovable(loc))
+			spans |= SPAN_COMMAND
+			message_mods[SAY_FAKE_COMMAND] = TRUE
```

Comment:
Rather than put this directly in the middle of saycode, this should be some kind of element that you can attach to objects which uses signals to inject the span 

You can use `connect_loc` for this - when an atom enters our object's turf register the say signal onto them. 
When our object moves away from the atom or our object leaves our object's turf, unregister the say signal. 
When the atom talks, add the span.